pymeasure/instruments/agilent/agilent8722ES.py

```diff
@@ -121,48 +131,51 @@ def set_IF_bandwidth(self, bandwidth):
                                  "8722ES exceeded")
 
     def set_averaging(self, averages):
-        """ Turns on averaging of a specific number between 0 and 999
-        """
-        if int(averages) > 999 or int(averages) < 0:
-            raise RangeException("Averaging must be in the range 0 to 999")
+        """Sets the number of averages and enables/disables averaging"""
+        averages = int(averages)
+        if not 1 <= averages <= 999:
+            assert RangeException("Invalid number of averages -", averages)
+        self.averages = averages
+        if averages > 1:
+            self.enable_averaging()
         else:
-            self.write("AVERO1")
-            self.write("AVERFACT%d" % int(averages))
+            self.disable_averaging()
 
     def disable_averaging(self):
-        """ Disables averaging """
+        """Disables averaging"""
         self.write("AVERO0")
 
+    def enable_averaging(self):
+        """Enables averaging"""
+        self.write("AVERO1")
+
     def is_averaging(self):
         """ Returns True if averaging is enabled """
         return self.ask("AVERO?") == '1\n'
 
-    def restart_averaging(self, averages):
-        if int(averages) > 999 or int(averages) < 0:
-            raise RangeException("Averaging must be in the range 0 to 999")
-        else:
-            self.write("NUMG%d" % averages)
-
-    def scan(self, averages=1, blocking=True, timeout=25, delay=0.1):
+    def scan(self):
         """ Initiates a scan with the number of averages specified and
-        blocks until the operation is complete if blocking is True
+        blocks until the operation is complete
         """
-        if averages == 1:
-            self.disableAveraging()
-            self.setSingleSweep()
-        else:
-            self.setAveraging(averages)
-            self.write("*CLS;SRE 4;ESNB 1;")
-            self.restartAveraging(averages)
-            if blocking:
-                self.adapter.wait_for_srq(timeout, delay)
-
-    def is_scan_complete():
-        pass  # TODO: Implement method for determining if the scan is completed
+        self.write("*CLS")
+        self.scan_single()
+        # All queries will block until the scan is done, so use NOOP? to check.
+        # These queries will time out after several seconds though,
+        # so query repeatedly until the scan finishes.
+        while True:
+            try:
+                self.ask("NOOP?")
+                break
```

Comment:
I would suggest placing the `break` in an `else` statement after the `except` statement.